app/src/main/java/com/github/se/signify/ui/screens/tutorial/TutorialScreen.kt

```diff
@@ -30,158 +20,191 @@ import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.font.FontWeight
-import androidx.compose.ui.text.style.TextAlign
-import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.text.style.TextDecoration
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import com.github.se.signify.R
 import com.github.se.signify.model.navigation.NavigationActions
 import com.github.se.signify.model.navigation.Screen
 import com.github.se.signify.ui.screens.home.HomeScreen
 
-private val overlayColor
-  @Composable get() = MaterialTheme.colorScheme.onBackground.copy(alpha = 0.8F)
-
 @Composable
 fun TutorialScreen(navigationActions: NavigationActions, onFinish: () -> Unit) {
   var step by remember { mutableIntStateOf(0) }
 
+  // Retrieving positions from CompositionLocal
+  val elementPositions = LocalElementPositions.current
+  val dictionaryPos = elementPositions["LetterDictionary"]
+  val cameraPos = elementPositions["CameraFeedbackButton"]
+  val exercisesPos = elementPositions["ExerciseList"]
+  val questsPos = elementPositions["QuestsButton"]
+  val quizPos = elementPositions["QuizButton"]
+  val feedbackPos = elementPositions["FeedbackButton"]
+
+  val overlayColor = MaterialTheme.colorScheme.onBackground.copy(alpha = 0.8F)
```

Comment:
Maybe you should refactor this function to reduce its cognitive complexity as you have too many nested if else